addons/stock/static/src/widgets/generate_serial.js

```diff
@@ -19,45 +19,59 @@ export class GenerateDialog extends Component {
     };
     setup() {
         this.size = 'md';
-        if (this.props.mode === 'generate') {
-            this.title = this.props.move.data.has_tracking === 'lot'
-            ? _t("Generate Lot numbers")
-            : _t("Generate Serial numbers");
-        } else {
-            this.title = this.props.move.data.has_tracking === 'lot' ? _t("Import Lots") : _t("Import Serials");
-        }
-
         this.nextSerial = useRef('nextSerial');
         this.nextSerialCount = useRef('nextSerialCount');
         this.totalReceived = useRef('totalReceived');
         this.keepLines = useRef('keepLines');
         this.lots = useRef('lots');
         this.orm = useService("orm");
+        if (this.props.mode === 'generate') {
+            if (this.isLotTracking) {
+                this.title = _t("Generate Lot numbers");
+                useAutofocus({refName: 'totalReceived'});
+            }
+            else {
+                this.title = _t("Generate Serial numbers");
+                useAutofocus({refName: 'nextSerialCount'});
+            }
+        } else {
+            this.title = this.isLotTracking ? _t("Import Lots") : _t("Import Serials");
+        }
         onWillStart(async () => {
             this.displayUOM = await user.hasGroup("uom.group_uom");
+            // Automatically generate lot/serial number.
+            if (this.props.mode === 'generate') {
+                await this._onGenerateCustomSerial();
+            }
         });
         onMounted(() => {
             if (this.props.mode === 'generate') {
                 this.nextSerialCount.el.value = this.props.move.data.product_uom_qty || 2;
-                if (this.props.move.data.has_tracking === 'lot') {
+                this.nextSerial.el.value = this.nextCustomSerialNumber;
```

Comment:
To avoid to set `undefined` as the value if `nextCustomSerialNumber` is undefined (can happen in `_onGenerateCustomSerial`, if there is no product's sequence, `nextCustomSerialNumber` won't be defined)
```suggestion
                if (this.nextCustomSerialNumber) {
                    this.nextSerial.el.value = this.nextCustomSerialNumber;
                }
```